src/libaktualizr/uptane/fetcher_test.cc

```diff
@@ -15,55 +16,6 @@ static std::string server = "http://127.0.0.1:";
 static std::string treehub_server = "http://127.0.0.1:";
 std::string sysroot;
 
-unsigned int num_events_DownloadPause = 0;
-void process_events_DownloadPauseResume(const std::shared_ptr<event::BaseEvent>& event) {
-  if (event->variant == "DownloadProgressReport") {
-    return;
-  }
-  switch (num_events_DownloadPause) {
-    case 0: {
-      EXPECT_EQ(event->variant, "DownloadPaused");
-      const auto pause_event = std::static_pointer_cast<event::DownloadPaused>(event);
-      EXPECT_EQ(pause_event->result, PauseResult::kNotDownloading);
-      break;
-    }
-    case 1: {
-      EXPECT_EQ(event->variant, "DownloadResumed");
-      const auto pause_event = std::static_pointer_cast<event::DownloadPaused>(event);
-      EXPECT_EQ(pause_event->result, PauseResult::kNotPaused);
-      break;
-    }
-    case 2: {
-      EXPECT_EQ(event->variant, "DownloadPaused");
-      const auto pause_event = std::static_pointer_cast<event::DownloadPaused>(event);
-      EXPECT_EQ(pause_event->result, PauseResult::kPaused);
-      break;
-    }
-    case 3: {
-      EXPECT_EQ(event->variant, "DownloadPaused");
-      const auto pause_event = std::static_pointer_cast<event::DownloadPaused>(event);
-      EXPECT_EQ(pause_event->result, PauseResult::kAlreadyPaused);
-      break;
-    }
-    case 4: {
-      EXPECT_EQ(event->variant, "DownloadResumed");
-      const auto resume_event = std::static_pointer_cast<event::DownloadResumed>(event);
-      EXPECT_EQ(resume_event->result, PauseResult::kResumed);
-      break;
-    }
-    case 5: {
-      EXPECT_EQ(event->variant, "DownloadResumed");
-      const auto resume_event = std::static_pointer_cast<event::DownloadResumed>(event);
-      EXPECT_EQ(resume_event->result, PauseResult::kNotPaused);
-      break;
-    }
-    default:
-      std::cout << "event #" << num_events_DownloadPause << " is: " << event->variant << "\n";
-      EXPECT_EQ(event->variant, "");
-  };
-  ++num_events_DownloadPause;
-}
-
```

Comment:
I understand why this is removed, but then we don't test that those events are actually created. I note that the events aren't actually mentioned in `actions.md`, and perhaps it isn't actually important, but is there a way we can still test these events easily?

Reply:
They'd have to be tested on `Aktualizr`'s side which might be hard...

Maybe we don't have to do it for every logic case, like it's done here, because the translation from pause/resume returns to an event is actually straightforward there.

Reply:
Yeah, and to be honest, emitting events for pause/resume was only done to be consistent. They aren't very useful since they just duplicate the function return values, and the function should return quickly.

Reply:
Ah indeed... But it makes some kind of sense if you want to observe (log) the systems through all these callbacks. I can try to come up with a quick test and if it's too convoluted, we could leave it like that?

Reply:
If you've got an idea, go for it! (And if it does result in new test coverage, might as well add the items to actions.md.)